### PR TITLE
Closes #51 Add categories

### LINKED ISSUE #51: Category screen
After picking a particular category, we should show the POP containing products from that category.

### PR DESCRIPTION
**Pull request by nikola-mladenovic**
Since the number of the view controllers and view models has grown significantly, I've rearranged them in separate groups, for each app section.

Until we have category assets available in the Sunrise project, the local images will be used.

TestFlight build is being processed.

### CODE REVIEW

**Review by cneijenhuis — APPROVED** [review]
👍  for the re-arrangement

File: Sunrise/ViewModels/Categories/CategoriesViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        // set the first root category to be an active one
+        activeCategories <~ rootCategories.producer.map { rootCategories in rootCategories.first != nil ? [rootCategories.first!] : [] }
+        activeRootCategoryName <~ activeCategories.producer.map { $0.first?.name?.localizedString }
+        activeCategories.combinePrevious(activeCategories.value).signal.observeValues { [weak self] previous, current in
+            self?.updateActiveCategory(from: previous, to: current)
+        }
+        // Use hardcoded images till we get assets on categories
+        backgroundImage <~ activeCategories.producer.map { (activeCategories: [Category]) -> UIImage? in
+            guard let rootCategoryName = activeCategories.first?.name?.localizedString else { return nil }
+            switch rootCategoryName {
```
Comment (nikola-mladenovic) [inline Sunrise/ViewModels/Categories/CategoriesViewModel.swift:67]: @cneijenhuis - currently we have 3 images, two specific to _Men_ and _Women_ categories, and I'm using the third one for all other categories.

Comment (cneijenhuis) [reply Sunrise/ViewModels/Categories/CategoriesViewModel.swift:67]: Yup, saw that :)

**Comment by cneijenhuis** [comment]
When going from Category to a POP, I'm always browsing the Online Store. Is that intentional?

**Comment by nikola-mladenovic** [comment]
That isn't intentional, working on fixing it

**Comment by nikola-mladenovic** [comment]
Fixed ReactiveSwift related leaks on POP and PDP, updated UI, my store POP sync from home and categories tab. Also added a fix for reservation notifications.

**Pull request merged**